pkg/operator/targetconfigcontroller/targetconfigcontroller.go

```diff
@@ -251,9 +259,35 @@ func managePod(ctx context.Context, client coreclientv1.ConfigMapsGetter, record
 	configMap.Data["pod.yaml"] = resourceread.WritePodV1OrDie(required)
 	configMap.Data["forceRedeploymentReason"] = operatorSpec.ForceRedeploymentReason
 	configMap.Data["version"] = version.Get().String()
+
+	startupMonitorPodKey, optionalStartupMonitor, err := manageOptionalStartupMonitorPod(infrastructureLister, operatorSpec, operatorImagePullSpec)
+	if err != nil {
+		return nil, false, fmt.Errorf("failed to apply an optional pod due to %v", err)
+	}
+	if optionalStartupMonitor != nil {
+		configMap.Data[startupMonitorPodKey] = resourceread.WritePodV1OrDie(optionalStartupMonitor)
+	}
 	return resourceapply.ApplyConfigMap(ctx, client, recorder, configMap)
 }
 
+func manageOptionalStartupMonitorPod(infrastructureLister configlistersv1.InfrastructureLister, operatorSpec *operatorv1.StaticPodOperatorSpec, operatorImagePullSpec string) (string, *corev1.Pod, error) {
+	infra, err := infrastructureLister.Get("cluster")
+	if err != nil && !apierrors.IsNotFound(err) {
+		// we got an error so without the infrastructure object we are not able to determine the type of platform we are running on
+		return "", nil, err
+	}
+	if infra.Status.ControlPlaneTopology != configv1.SingleReplicaTopologyMode {
+		return "", nil, nil
+	}
+
+	appliedStartupMonitorPodTemplate, err := startupmonitor.ManagePodTemplate(operatorSpec, []string{"cluster-kube-apiserver-operator", "startup-monitor"}, operatorclient.TargetNamespace, "kube-apiserver", operatorImagePullSpec)
```

Comment:
same here: ManagePodTemplate does not do any API action, does it? Better call it (in library-go) generate.